query.py

```diff
@@ -0,0 +1,272 @@
+
+            # Exclude results of tests with fewer than 10 round trip time samples,
+            # because there are not enough samples to accurately estimate the RTT.
+            if metric == 'minimum_rtt':
+                valid_metric_conditional.append(
+                    'web100_log_entry.snap.CountRTT > %u' % QueryConditionals.MIN_RTT_SAMPLES)
+        else:
+            # Must receive at least the minimum number of bytes.
+            valid_metric_conditional.append(
+                'web100_log_entry.snap.HCThruOctetsReceived >= %u' % QueryConditionals.MIN_BYTES)
+            # Must last for at least the minimum test duration.
+            valid_metric_conditional.append('web100_log_entry.snap.Duration >= %u' % QueryConditionals.MIN_DURATION)
+            # Must not exceed the maximum test duration.
+            valid_metric_conditional.append('web100_log_entry.snap.Duration < %u' % QueryConditionals.MAX_DURATION)
+        metric_dict[metric] = '\n\tAND '.join(valid_metric_conditional)
+        return metric_dict
+
+    def get_conditional_dict(self, metric):
+        if metric not in METRICS:
+            raise NotImplementedError()
+        metric_dict = self._conditional_dict.copy()
+        metric_dict.update(self._create_metric_specific_dict(metric))
+        return metric_dict
+
+
+class SubQueryGenerator(object):
+
+    def __init__(self, metric, start_time, end_time, client_ip_blocks):
+        self._conditionals = QueryConditionals(start_time, end_time, client_ip_blocks).get_conditional_dict(metric)
+        self._metric = metric
+        self._query = self._create_query_string()
+
+    @property
+    def query(self):
+        return self._query
+
+    def _create_query_string(self):
+        built_query_format = ('SELECT\n'
+                                '\t{select_clauses}\n'
+                              'FROM\n'
+                                '\t{table}\n'
+                              'WHERE\n'
+                                '\t{conditional_list}')
+
+        conditional_list_string = ''
+        conditional_list_string += self._conditionals['data_direction']
+        conditional_list_string += '\n\t AND %s' % (
+            self._conditionals[self._metric])
+        conditional_list_string += '\n\tAND %s' % self._conditionals['log_time']
+
+        client_ip_blocks_joined = ' OR\n\t\t'.join(self._conditionals[
+            'client_ip_blocks'])
+        conditional_list_string += '\n\tAND (%s)' % client_ip_blocks_joined
+
+        built_query_string = built_query_format.format(
+            select_clauses=self._create_select_clauses(self._metric),
+            table='plx.google:m_lab.ndt.all',
+            conditional_list=conditional_list_string)
+
+        return built_query_string
+
+    def _create_select_clauses(self, metric):
+        if metric == 'download':
+          return ('web100_log_entry.log_time AS timestamp,\n'
```

Comment:
I just found out about `textwrap.dedent()` which makes multi-line strings no longer awful. So you could also do:
```python
    return texwrap.dedent('''\
        web100_log_entry.log_time AS timestamp,
            8 * (web100_log_entry.snap.HCThruOctetsAcked /
                 (web100_log_entry.snap.SndLimTimeRwin +
                  web100_log_entry.snap.SndLimTimeCwnd +
                  web100_log_entry.snap.SndLimTimeSnd)) AS download_mbps''')
```

And it will do what you want it to. I think everything is a lot more readable without all those \t and \n characters in it.